Assets/Scripts/EnemyPatrol.cs

```diff
@@ -0,0 +1,239 @@
+
+    public short enemyHealth;
+
+    private bool hittingWall, notEdge, playerInRange, moveRight, 
+        isAttacking, hasAttacked, attackedOnRight, canMove, isDamagable, beingKnockedBack;
+    
+    PlayerController playerController;
+
+    EnemyHealth EnemyHealthController;
+
+    CircleCollider2D enemyAttackPoint;
+
+    [SerializeField]
+    Transform pathCheck;
+
+    [SerializeField]
+    Transform edgeCheck;
+    
+    [SerializeField]
+    LayerMask whatIsWall, whatIsPlayer;
+    
+    Rigidbody2D enemyRigidBody;
+
+    Animator anim;
+
+    [SerializeField]
+    PhysicsMaterial2D knockbackPM, patrolPM;
+
+    Collider2D enemyPrimaryCollision;
+
+    // Use this for initialization
+    void Start() {
+        SetupObject();
+    }
+
+    //Gather and Set Necessary Components, GameObjects, Values, and States
+    void SetupObject()
+    {
+        // Components
+        anim = GetComponent<Animator>();
+        enemyRigidBody = GetComponent<Rigidbody2D>();        
+        enemyPrimaryCollision = GetComponent<CapsuleCollider2D>();
+        enemyAttackPoint = GetComponentInChildren<CircleCollider2D>();
+        EnemyHealthController = GetComponentInChildren<EnemyHealth>();
+
+        //Values
+        EnemyHealthController.UpdateHealth(enemyHealth);
+
+        // States
+        hasAttacked = false;
+        enemyAttackPoint.enabled = false;
+        canMove = true;
+        isDamagable = true;
+        beingKnockedBack = false;
+    }
+
+    //FixedUpdate
+    void FixedUpdate()
+    {
+        PositionChecks();
+        if(canMove)
+            Patrol();     
+    }
+
+    // Update is called once per frame
+    void Update () {
+        MeleeCheck();
+        anim.SetFloat("Speed", Mathf.Abs(enemyRigidBody.velocity.x));
+
+        UpdatePM();
+
+        if (anim.GetCurrentAnimatorStateInfo(0).IsName("Anim_Bandit_Attack"))
+            enemyAttackPoint.enabled = true;
+        else
+            enemyAttackPoint.enabled = false;
+    }
+
+    void UpdatePM()
+    {
+        if (beingKnockedBack)
+            enemyPrimaryCollision.sharedMaterial = knockbackPM;
+        else
+            enemyPrimaryCollision.sharedMaterial = patrolPM;
+    }
+
+    //Handle movement of Enemies to let them 'patrol' the area
+    private void Patrol()
+    {
+        if (hittingWall || notEdge)
+            moveRight = !moveRight;
+
+        if (moveRight)
+        {
+            transform.localScale = new Vector3(1, 1, 1);
+            enemyRigidBody.velocity = new Vector2(moveSpeed, enemyRigidBody.velocity.y);
+        }
+        else
+        {
+            transform.localScale = new Vector3(-1, 1, 1);
+            enemyRigidBody.velocity = new Vector2(-moveSpeed, enemyRigidBody.velocity.y);
+        }
+
+    }
+
+    //Check position of the enemy based on the associated Transform.GameObjects
+    private void PositionChecks()
+    {
+        //WALL CHECK
+        hittingWall = Physics2D.OverlapCircle(pathCheck.position,
```

Comment:
I'd extract each of these into their own methods like `UpdateIsHittingWall()` with summaries that explain what they do.